McCalib/include/geometrytools.hpp

```diff
@@ -52,3 +52,8 @@ void projectPointsWithDistortion(std::vector<cv::Point3f> object_pts,
                                  cv::Mat distortion_vector,
                                  std::vector<cv::Point2f> &repro_pts,
                                  int distortion_type);
+cv::Mat convertRotationMatrixToQuaternion(cv::Mat R);
+cv::Mat convertQuaternionToRotationMatrix(const std::array<double, 4> &q);
+cv::Mat getAverageRotation(std::vector<double> &r1, std::vector<double> &r2,
+                           std::vector<double> &r3,
+                           const bool use_quaternion_averaging = true);
```

Comment:
That's great that we have this Boolean to switch from median to average rotation. While the median computation is not formally/theoretically correct and certainly leads to inaccurate results, it might be interesting to let the user select the rotation averaging technique they want; just in case. My guess is that median should still be a bit more robust against outliers than the average (not sure if it is a common scenario as we have plenty of intermediate stage which are designed to avoid such outliers). 
Anyway, I will just adding this option in the config file if possible

Reply:
Added as an option in a204074575e428507075ade706822f062d4754e8